From a958795a964cc6c07b172f5f9ea7dec990933bee Mon Sep 17 00:00:00 2001
From: Bo Gan <ganb@vmware.com>
Date: Mon, 14 Sep 2020 16:19:58 -0700
Subject: [PATCH] i40e: Remove read_barrier_depends in favor of READ_ONCE

read_barrier_depends() is removed and implied by READ_ONCE.
Architecture (Alpha) that needs read_barrier_depends() enforce it
via specializing READ_ONCE.

Signed-off-by: Bo Gan <ganb@vmware.com>
---
 src/i40e_main.c | 5 +----
 src/i40e_txrx.c | 5 +----
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/src/i40e_main.c b/src/i40e_main.c
index ac35887..33f7fc3 100644
--- a/src/i40e_main.c
+++ b/src/i40e_main.c
@@ -4648,15 +4648,12 @@ static bool i40e_clean_fdir_tx_irq(struct i40e_ring *tx_ring, int budget)
 	i -= tx_ring->count;
 
 	do {
-		struct i40e_tx_desc *eop_desc = tx_buf->next_to_watch;
+		struct i40e_tx_desc *eop_desc = READ_ONCE(tx_buf->next_to_watch);
 
 		/* if next_to_watch is not set then there is no work pending */
 		if (!eop_desc)
 			break;
 
-		/* prevent any other reads prior to eop_desc */
-		read_barrier_depends();
-
 		/* if the descriptor isn't done, no work yet to do */
 		if (!(eop_desc->cmd_type_offset_bsz &
 		      cpu_to_le64(I40E_TX_DESC_DTYPE_DESC_DONE)))
diff --git a/src/i40e_txrx.c b/src/i40e_txrx.c
index e3eb496..39c1c72 100644
--- a/src/i40e_txrx.c
+++ b/src/i40e_txrx.c
@@ -981,15 +981,12 @@ static bool i40e_clean_tx_irq(struct i40e_vsi *vsi,
 	tx_head = I40E_TX_DESC(tx_ring, i40e_get_head(tx_ring));
 
 	do {
-		struct i40e_tx_desc *eop_desc = tx_buf->next_to_watch;
+		struct i40e_tx_desc *eop_desc = READ_ONCE(tx_buf->next_to_watch);
 
 		/* if next_to_watch is not set then there is no work pending */
 		if (!eop_desc)
 			break;
 
-		/* prevent any other reads prior to eop_desc */
-		read_barrier_depends();
-
 		i40e_trace(clean_tx_irq, tx_ring, tx_desc, tx_buf);
 		/* we have caught up to head, no work left to do */
 		if (tx_head == tx_desc)
-- 
2.7.4

